nixos/modules/services/monitoring/chronograf.nix

```diff
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.chronograf;
+in
+{
+
+  options.services.chronograf = {
+    enable = mkEnableOption "chronograf server";
+
+    port = mkOption {
+      default = 8888;
+      type = types.int;
+      description = ''
+        The port where chronograf is going to be listening.
+      '';
+    };
+
+    dataDir = mkOption {
+      default = "/var/db/chronograf";
+      type = types.str;
+      description = ''
+        Data directory for chronograf data files.
+      '';
+    };
+
+    influxdbURL = mkOption {
+      default = "";
+      type = types.str;
+      description = ''
+        The URL of the InfluxDB that Chronograf is going to connect to.
+      '';
+      example ="http://localhost:8086";
+    };
+
+    kapacitorURL = mkOption {
+      default = "";
+      type = types.str;
+      description = ''
+        The URL of the Kapacitor that Chronograf is going to connect to.
+      '';
+    };
+
+    reporting = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Disable reporting of usage stats (os,arch,version,cluster_id,uptime) once every 24hr.
+      '';
+    };
+
+    extraConfig = mkOption {
+      default = {};
+      type = types.attrsOf types.str;
+      description = ''
+        Chronograf can be configured using environment variables. For more information,
+        check https://docs.influxdata.com/chronograf/v1.7/administration/config-options.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}' 0770 chronograf chronograf - -"
+    ];
+
+    systemd.services.chronograf = {
```

Comment:
Is it possible to reuse the [upstream service file](https://github.com/influxdata/chronograf/blob/0568cac0406fe8c15320f9864e2102635aff2cf0/etc/scripts/chronograf.service) after patching it?